drivers/Aqara/aqara-cube/src/init.lua

```diff
@@ -0,0 +1,127 @@
+
+local is_aqara_products = function(opts, driver, device, ...)
+  return device:get_manufacturer() == FINGERPRINTS.mfr and device:get_model() == FINGERPRINTS.model
+end
+
+local callback_timer = function(device)
+  return function()
+    device:emit_event(cubeAction.cubeAction("noAction"))
+  end
+end
+
+local function reset_thread(device)
+  local timer = device:get_field(CUBEACTION_TIMER)
+  if timer then
+    device.thread:cancel_timer(timer)
+    device:set_field(CUBEACTION_TIMER, nil)
+  end
+  device:set_field(CUBEACTION_TIMER, device.thread:call_with_delay(CUBEACTION_TIME, callback_timer(device)))
+end
+
+local function data_handler(driver, device, value, zb_rx)
+  local val = value.value
+  if val == 0x0000 then -- Shake
+    reset_thread(device)
+    device:emit_event(cubeAction.cubeAction("shake"))
+  elseif val == 0x0004 then -- hold
+    reset_thread(device)
+    device:emit_event(cubeAction.cubeAction("pickUpAndHold"))
+  elseif val & 0x0400 == 0x0400 then -- Flip to side
+    local faceNum = val & 0x0007
+    reset_thread(device)
+    device:emit_event(cubeAction.cubeAction(cubeFlipToSideVal[faceNum + 0x1]))
+  end
+end
+
+local function rotate_handler(driver, device, value, zb_rx)
+  -- Rotation
+  reset_thread(device)
+  device:emit_event(cubeAction.cubeAction("rotate"))
+end
+
+local function face_handler(driver, device, value, zb_rx)
+  local faceNum = value.value
+  device:emit_event(cubeFace.cubeFace(cubeFaceVal[faceNum + 1]))
+end
+
+local function device_init(driver, device)
+  local power_configuration = {
+    cluster = PowerConfiguration.ID,
+    attribute = PowerConfiguration.attributes.BatteryVoltage.ID,
+    minimum_interval = 30,
+    maximum_interval = 3600,
+    data_type = PowerConfiguration.attributes.BatteryVoltage.base_type,
+    reportable_change = 1
+  }
+
+  battery_defaults.build_linear_voltage_init(2.6, 3.0)(driver, device)
+
+  device:add_configured_attribute(power_configuration)
+  device:add_monitored_attribute(power_configuration)
+end
+
+local function device_added(self, device)
+  -- Set private attribute
+  device:send(cluster_base.write_manufacturer_specific_attribute(device,
+    PRI_CLU, PRI_ATTR, MFG_CODE, data_types.Uint8, 1))
+
+  device:send(cluster_base.write_manufacturer_specific_attribute(device,
+    PRI_CLU, CUBE_MODE, MFG_CODE, data_types.Uint8, 1))
+  device:emit_event(cubeAction.cubeAction("noAction"))
+  device:emit_event(cubeFace.cubeFace("face1Up"))
+  device:emit_event(capabilities.battery.battery(100))
+end
+
+-- [[ register ]]
+local aqara_cube_t1_pro_handler = {
+  NAME = "Aqara Cube T1 Pro",
+  zigbee_handlers = {
+    attr = {
+      [EVENT_CLU] = {
+        [ACTION_ATTR] = data_handler
+      },
+      [ROTATE_CLU] = {
+        [ACTION_ATTR] = rotate_handler,
+      },
+      [PRI_CLU] = {
+        [FACE_ATTR] = face_handler
+      }
+    }
+  },
+  lifecycle_handlers = {
+    init = device_init,
+    added = device_added
+  },
+  can_handle = is_aqara_products
```

Comment:
as a base driver, you do not need to supply a `can_handle` function

Reply:
It was removed can_handle.